HW04/anastasia_evsihneyeva/lib/mentor.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class Mentor
+  attr_reader :name, :surname, :homework, :self
+  attr_accessor :notification_list, :all_hw
+
+  def initialize(name:, surname:)
+    @name = name
+    @surname = surname
+    @subscriptions = []
+    @all_hw = []
+    @notification_list = []
+  end
+
+  def notifications
+    notification_list.each(&:output)
+  end
+
+  def notify(notification)
+    notification_list << notification
+  end
+
+  def mark_as_read!
+    notification_list.clear
+  end
+
+  def add_homework(title:, description:, student:)
+    homework = Homework.new(title, description, student, self)
+    homework.student.notify(Notification.new("Homework #{title} was added to #{student.name} by mentor #{name}"))
```

Comment:
It seems to me that the student should not notify himself when the mentor adds homework, make this method on the mentor and call it to add notification to the student